src/components/AnonymousChat/BaseChatSystem.tsx

```diff
@@ -21,17 +21,25 @@ const styles = StyleSheet.create({
   }
 });
 
-const BaseSystemChat: React.FC<BaseChatItemComponentProps> = ({item, index}) => {
+const BaseSystemChat: React.FC<BaseChatItemComponentProps> = ({
+  item,
+  index,
+  messageSingle,
+  componentType = 'GROUP'
+}) => {
   const {handleTextSystem, splitSystemMessage} = useChatUtilsHook();
 
   const messageText = () => {
-    return splitSystemMessage(handleTextSystem(item))?.map((message, key) => {
-      return (
-        <Text key={key} style={styles.textSystem}>
-          {message}.
-        </Text>
-      );
-    });
+    if (componentType === 'GROUP') {
+      return splitSystemMessage(handleTextSystem(item))?.map((message, key) => {
+        return (
+          <Text key={key} style={styles.textSystem}>
+            {message}.
+          </Text>
+        );
+      });
+    }
+    return <Text style={styles.textSystem}>{messageSingle}.</Text>;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a `componentType` prop and a `messageSingle` prop. It seems like the `componentType` is used to determine how to render the message text. If the `componentType` is 'GROUP', it splits the system message and maps over each part, rendering them separately. If not, it simply renders the `messageSingle`. 

However, there's no fallback or default case if `componentType` is neither 'GROUP' nor any other expected value. This could potentially lead to unexpected behavior or errors. Consider adding a default case in your conditional logic.

```diff
   const messageText = () => {
    if (componentType === 'GROUP') {
      return splitSystemMessage(handleTextSystem(item))?.map((message, key) => {
        return (
          <Text key={key} style={styles.textSystem}>
            {message}.
          </Text>
        );
      });
    }
+    else if (messageSingle) {
    return <Text style={styles.textSystem}>{messageSingle}.</Text>;
+    } else {
+      // Default case
+      return null;
+    }
   };
```
This way, if neither condition is met, the function will return `null`, preventing potential errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->